x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,404 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CloudSetup } from '@kbn/cloud-plugin/server';
+import { DeprecationsDetails } from '@kbn/core-deprecations-common';
+import { GetDeprecationsContext, RegisterDeprecationsConfig } from '@kbn/core-deprecations-server';
+
+import { hasFleetServers } from '@kbn/fleet-plugin/server/services/fleet_server';
+import { isAgentlessEnabled } from '@kbn/fleet-plugin/server/services/utils/agentless';
+import { i18n } from '@kbn/i18n';
+import { Connector, fetchConnectors } from '@kbn/search-connectors';
+
+import { ConfigType } from '..';
+
+export const getRegisteredDeprecations = (
+  config: ConfigType,
+  cloud: CloudSetup,
+  docsUrl: string
+): RegisterDeprecationsConfig => {
+  return {
+    getDeprecations: async (ctx: GetDeprecationsContext) => {
+      const hasAgentless = isAgentlessEnabled();
+      const hasFleetServer = await hasFleetServers(
+        ctx.esClient.asInternalUser,
+        ctx.savedObjectsClient
+      );
+      return [
+        ...getEnterpriseSearchNodeDeprecation(config, cloud, docsUrl),
+        ...(await getCrawlerDeprecations(ctx, docsUrl)),
+        ...(await getNativeConnectorDeprecations(
+          ctx,
+          hasAgentless,
+          hasFleetServer,
+          cloud,
+          docsUrl
+        )),
+      ];
```

Comment:
nit: not required but could be more readable and would give more options for handling errors if we need to do that.

```suggestion
      const entSearchDetails = getEnterpriseSearchNodeDeprecation(config, cloud, docsUrl);
      // this will run the next two async calls together instead of in series.
      const [crawlerDetails, nativeConnectorsDetails] = await Promise.all([
        getCrawlerDeprecations(ctx, docsUrl),
        getNativeConnectorDeprecations(
          ctx,
          hasAgentless,
          hasFleetServer,
          cloud,
          docsUrl
        ),
      ]);
      
      return [
        ...entSearchDetails,
        ...crawlerDetails,
        ...nativeConnectorsDetails,
      ];
```